### DISCUSSION TITLE
Timeout?

### DISCUSSION THREAD

**Discussion by Elliot01Sepiol** [Qmirdev/EthelMC#5] [Q&A] [answered]
How to set timeouts for users attempting to run bot commands?

**Comment by Elliot01Sepiol** [comment]
> How to set timeouts for users attempting to run bot commands?

I mean in a way to prevent users spamming the bot commands in chat

**Comment by q31r** [comment, marked as answer]
This function sets a cooldown of 5 seconds for users to prevent spamming the commands _(!help command for example in here)_. If the user tries to use the command within the cooldown period, their message will be deleted ;)
```python
cooldowns = {} 
def check_cooldown(ctx):
  now = time.time()
  user_id = ctx.author.id
  if user_id in cooldowns and cooldowns[user_id] > now:
    return False
  cooldowns[user_id] = now + 5 
  return True

@client.command()
async def help(ctx):
  if not check_cooldown(ctx):
    await ctx.message.delete()
    return
  else:
    response_message = f"Hi <@{ctx.author.id}>, {SERVER_NAME} bot is here! If you have any questions/issues related to the Minecraft or Discord Server, you can ask in {DISCORD_QA_CHANNEL} or create a ticket at {DISCORD_TICKET_CHANNEL} and one of our staff members will reply as soon as possible!"
    await ctx.send(response_message)
```

**Reply by Elliot01Sepiol** [reply]
Thank you for providing exactly what I was looking for. It would be fantastic if you could add this feature for every command in the repository.

**Reply by q31r** [reply]
no problem, i will add this in the next update 😃
